### PR TITLE
techdebt: [CDS-78650]: Remove reflection from consumers

### PR DESCRIPTION
**Pull request by hextechpal**
## Describe your changes
Remove reflection from consumers to optimize on CPU. The triggers consumers were not setup correctly hence refactored them a little big to follow the pattern
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52361)

### CODE REVIEW

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/event/pollingevent/PollingEventStreamListener.java` line 41 at r1](https://reviewable.io/reviews/harness/harness-core/52361#-NdFSNVN9Xg2bw_xjaDe:-NdFSNVN9Xg2bw_xjaDf:b-uxz426) ([raw file](https://github.com/harness/harness-core/blob/f4f35ede7d7155d2dfa93415a1efcb93b90b7800/pipeline-service/service/src/main/java/io/harness/pms/event/pollingevent/PollingEventStreamListener.java#L41)):*
> ```Java
> @Slf4j
> @OwnedBy(PIPELINE)
> public class PollingEventStreamListener extends PmsAbstractMessageListener<FacilitatorEvent, FacilitatorEventHandler> {
> ```

why is faciliatorEvent coming in this?

**Review by hextechpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52361)* status: 26 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/event/pollingevent/PollingEventStreamListener.java` line 41 at r1](https://reviewable.io/reviews/harness/harness-core/52361#-NdFSNVN9Xg2bw_xjaDe:-NdFTq9K3e6Vbd4rsOdz:bnlanyw) ([raw file](https://github.com/harness/harness-core/blob/f4f35ede7d7155d2dfa93415a1efcb93b90b7800/pipeline-service/service/src/main/java/io/harness/pms/event/pollingevent/PollingEventStreamListener.java#L41)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

why is faciliatorEvent coming in this?
</blockquote></details>

Earlier it was written like this and everything was overridden corrected these in my PRs, same for other Listeners as well

**Review by archit-harness — APPROVED** [review]
> Reviewed 26 of 26 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52361)* status: 26 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/events/base/PmsAbstractMessageListener.java` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/52361#-NdFTCraEzSPgLqJTfcU:-NdFTCraEzSPgLqJTfcV:bdanukf) ([raw file](https://github.com/harness/harness-core/blob/b68ea232e6fe06ba3494015378fc8898e1c3f3ef/884-pms-commons/src/main/java/io/harness/pms/events/base/PmsAbstractMessageListener.java#L55)):*
> ```Java
>       processMessage(entity, message.getMessage().getMetadataMap(), issueTimestamp, readTs);
>     } catch (InvalidProtocolBufferException ex) {
>       log.error("Cannot decode bytes into object");
> ```

can we add messageId in log message and ex getMessage

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/service/TriggerWebhookExecutionServiceV2.java [hunk 1]
```diff
@@ -8,7 +8,8 @@
 package io.harness.pms.triggers.webhook.service;
 
 import io.harness.eventsframework.webhookpayloads.webhookdata.WebhookDTO;
+import io.harness.pms.sdk.execution.events.PmsCommonsBaseEventHandler;
 
-public interface TriggerWebhookExecutionServiceV2 {
+public interface TriggerWebhookExecutionServiceV2 extends PmsCommonsBaseEventHandler<WebhookDTO> {
```
Comment (sribalij) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/service/TriggerWebhookExecutionServiceV2.java:13]: **nit**: Should we name this as a `TriggerWebhookExecutionServiceHandlerV2`

**Review by sribalij — APPROVED** [review]

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/service/TriggerWebhookExecutionServiceV2.java (hunk 1 above)
Comment (hextechpal) [reply pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/service/TriggerWebhookExecutionServiceV2.java:13]: Would do it in a separate PR, trying to keep the PR focussed

**Review by hextechpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52361)* status: 26 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/events/base/PmsAbstractMessageListener.java` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/52361#-NdFTCraEzSPgLqJTfcU:-NdUGQURE4Emd3aNOART:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b68ea232e6fe06ba3494015378fc8898e1c3f3ef/884-pms-commons/src/main/java/io/harness/pms/events/base/PmsAbstractMessageListener.java#L55)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can we add messageId in log message and ex getMessage
</blockquote></details>

Done.

**Comment by hextechpal** [comment]
trigger uts

**Review by archit-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52361)* status: 26 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): sto.

**Comment by hextechpal** [comment]
trigger uts

**Comment by hextechpal** [comment]
trigger ut6

**Comment by hextechpal** [comment]
trigger ut7

**Review by archit-harness — APPROVED** [review]

**Review by r0610205 — APPROVED** [review]

**Comment by hextechpal** [comment]
trigger review

**Comment by hextechpal** [comment]
trigger ut7

**Comment by hextechpal** [comment]
trigger ut7

**Pull request merged**